src/app/core/workbox/workbox.service.ts

```diff
@@ -93,20 +93,28 @@ export class WorkboxService {
         this.priorityTabUpdate.next(1);
     }
 
-    openTab(tab, persistToRecentApps: boolean = true, syncState = true) {
+    openTab(tab, persistToRecentApps: boolean = true, syncState = true, replaceExistingIfExists = false) {
 
         const {tabs} = this.extractValues();
 
         // When opening an app, we use id with revision number because we can have cases when we can open the same app
         // different revisions (when we publish a local file with existing id, new tab is open ...)
-        const foundTab = tabs.find(existingTab => existingTab.id === tab.id);
+        const foundTabIndex = tabs.findIndex(existingTab => existingTab.id === tab.id);
+        const foundTab = tabs[foundTabIndex];
 
-        if (foundTab) {
-            this.activateTab(foundTab);
-            return;
+        if (!replaceExistingIfExists) {
+            if (foundTab) {
+                this.activateTab(foundTab);
+                return;
+            }
+
+            this.tabs.next(tabs.concat(tab));
+        } else if (foundTab) {
+            tabs.splice(foundTabIndex, 1)
+            tabs.splice(foundTabIndex, 0, tab);
```

Comment:
tabs.splice(foundTabIndex, 1)
tabs.splice(foundTabIndex, 0, tab);

is the same as:
tabs.splice(foundTabIndex, 1, tab)